application/actions/meetings.go

```diff
@@ -1,11 +1,183 @@
 	"github.com/silinternational/wecarry-api/api"
 	"github.com/silinternational/wecarry-api/models"
+	"time"
 )
 
-func convertMeeting(meeting models.Meeting) api.Meeting {
+// swagger:operation GET /events Events ListEvents
+//
+// gets a list of meetings
+//
+// ---
+// responses:
+//   '200':
+//     description: meetings list
+//     schema:
+//       "$ref": "#/definitions/Meetings"
+func meetingsList(c buffalo.Context) error {
+	cUser := models.CurrentUser(c)
+	tx := models.Tx(c)
+	meetings := models.Meetings{}
+	if err := meetings.FindOnOrAfterDate(tx, time.Now().UTC()); err != nil {
+		return reportError(c, api.NewAppError(err, api.ErrorMeetingsGet, api.CategoryInternal))
+	}
+
+	output, err := convertMeetings(c, meetings, cUser)
+	if err != nil {
+		return reportError(c, api.NewAppError(err, api.ErrorMeetingsConvert, api.CategoryInternal))
+	}
+
+	return c.Render(200, render.JSON(output))
+}
+
+// converts list of model.Meeting into list of api.Meeting
+func convertMeetings(ctx context.Context, meetings []models.Meeting, user models.User) ([]api.Meeting, error) {
+	output := make([]api.Meeting, len(meetings))
+
+	for i, m := range meetings {
+		var err error
+		output[i], err = convertMeeting(ctx, m, user)
+		if err != nil {
+			return []api.Meeting{}, err
+		}
+	}
+
+	return output, nil
+}
+
+// converts a model.Meeting into api.Meeting
+func convertMeeting(ctx context.Context, meeting models.Meeting, user models.User) (api.Meeting, error) {
+	output := convertMeetingAbridged(meeting)
+
+	createdBy, err := loadMeetingCreatedBy(ctx, meeting)
+	if err != nil {
+		return api.Meeting{}, err
+	}
+	output.CreatedBy = createdBy
+
+	imageFile, err := loadMeetingImageFile(ctx, meeting)
+	if err != nil {
+		return api.Meeting{}, err
+	}
+	output.ImageFile = imageFile
+
+	location, err := loadMeetingLocation(ctx, meeting)
+	if err != nil {
+		return api.Meeting{}, err
+	}
+	output.Location = location
+
+	participants, err := loadMeetingParticipants(ctx, meeting, user)
+	if err != nil {
+		return api.Meeting{}, err
+	}
+	output.Participants = participants
+
+	return output, nil
+}
+
+func loadMeetingCreatedBy(ctx context.Context, meeting models.Meeting) (api.User, error) {
+	createdBy, err := meeting.GetCreator(models.Tx(ctx))
+	if err != nil {
+		return api.User{}, errors.New("loading meeting creator, " + err.Error())
+	}
+
+	outputCreatedBy, err := convertUser(ctx, *createdBy)
+	if err != nil {
+		err = errors.New("error converting meeting created_by user: " + err.Error())
+		return api.User{}, err
+	}
+	return outputCreatedBy, nil
+}
+
+func loadMeetingImageFile(ctx context.Context, meeting models.Meeting) (*api.File, error) {
+	imageFile, err := meeting.ImageFile(models.Tx(ctx))
+	if err != nil {
+		err = errors.New("error converting meeting image file: " + err.Error())
+		return nil, err
+	}
+
+	if imageFile == nil {
+		return nil, nil
+	}
+
+	var outputImage api.File
+	if err := api.ConvertToOtherType(imageFile, &outputImage); err != nil {
```

Comment:
Seems like a `convertFile` might be helpful here (and elsewhere)

Reply:
+1